great_expectations/util.py

```diff
@@ -689,6 +752,89 @@ def lint_code(code):
         return code
 
 
+def filter_properties_dict(
+    properties: dict,
+    keep_fields: Optional[list] = None,
+    delete_fields: Optional[list] = None,
+    clean_empty: Optional[bool] = True,
+    inplace: Optional[bool] = False,
+) -> Optional[dict]:
+    """Filter the entries of the source dictionary according to directives concerning the existing keys and values.
+
+    Args:
+        properties: source dictionary to be filtered according to the supplied filtering directives
+        keep_fields: list of keys that must be retained, with the understanding that all other entries will be deleted
+        delete_fields: list of keys that must be deleted, with the understanding that all other entries will be retained
+        clean_empty: If True, then in addition to other filtering directives, delete entries, whose values are Falsy
+        inplace: If True, then modify the source properties dictionary; otherwise, make a copy for filtering purposes
+
+    Returns:
+        The (possibly) filtered properties dictionary (or None if no entries remain after filtering is performed)
+    """
+    if keep_fields and delete_fields:
+        raise ValueError(
+            "Only one of keep_fields and delete_fields filtering directives can be specified."
+        )
+
+    if not inplace:
+        properties = copy.deepcopy(properties)
+
+    keys_for_deletion: list = []
+
+    if keep_fields:
+        keys_for_deletion.extend(
+            [key for key, value in properties.items() if key not in keep_fields]
+        )
+
+    if delete_fields:
+        keys_for_deletion.extend(
+            [key for key, value in properties.items() if key in delete_fields]
+        )
+
+    if clean_empty:
+        keys_for_deletion.extend(
+            [
+                key
+                for key, value in properties.items()
+                if not (
+                    (keep_fields and key in keep_fields)
+                    or is_numeric(value=value)
+                    or value
+                )
+            ]
+        )
+
+    keys_for_deletion = list(set(keys_for_deletion))
+
+    for key in keys_for_deletion:
+        del properties[key]
+
+    if inplace:
+        return None
+
+    return properties
+
+
+def is_numeric(value: Any) -> bool:
+    return value is not None and (is_int(value) or is_float(value))
+
+
+def is_int(value: Any) -> bool:
+    try:
+        num: int = int(value)
+    except ValueError:
+        return False
+    return True
+
+
+def is_float(value: Any) -> bool:
+    try:
+        num: float = float(value)
+    except ValueError:
+        return False
+    return True
+
```

Comment:
Thank you for these. I may have written my own version of these in either the `sorters.py` or `partition_query.py` files, and I agree that `util.py` is the right place for them to live.

I'll adjust the other code to use these methods, and move/consolidate any repeating methods.

Reply:
@Shinnnyshinshin If possible, let us wait for this PR to be merged.  Thanks!

Reply:
absolutely